### PR TITLE
[#9] Ordering of defdelegate

### LINKED ISSUE #9: Ordering of `defdelegate`
## Why

According to the Elixir convention on [Compass](https://nimblehq.co/compass/development/code-conventions/elixir/). We need to create a custom Credo rule to check that the defdelegate is placed before any public or private function.

image

## Acceptance criteria

When running Credo check, raises a warning if there is any `defdelegate` present after the first `def` or `defp` block.

### PR DESCRIPTION
**Pull request by hanam1ni**
- Closes #9 

## What happened

Add credo check for defdelegate must be placed in the module before any functions
 
## Proof Of Work

Run Credo check on the following module

image

The issue is reported

image

### CODE REVIEW

File: priv/.credo.exs [hunk 1]
```diff
@@ -3,7 +3,7 @@
     %{
       name: "default",
       checks: [
-        # TODO: Add custom checks module
+        {CompassCredoPlugin.Checks.DefdelegateOrder, []}
```
Comment (liamstevens111) [inline priv/.credo.exs:6]:
```suggestion
        {CompassCredoPlugin.Check.DefdelegateOrder, []}
```
Don't forget in this place too 😛

File: lib/check/defdelegate_order.ex [hunk 2]
```diff
@@ -0,0 +1,71 @@
+defmodule CompassCredoPlugin.Check.DefdelegateOrder do
+  use Credo.Check,
+    base_priority: :normal,
+    category: :readability,
+    explanations: [
+      check: """
+        Prefer placing defdelegate functions the functions.
```
Comment (liamstevens111) [inline lib/check/defdelegate_order.ex:7]:
```suggestion
        Prefer placing defdelegate before public and private functions.
```

Comment (hanam1ni) [reply lib/check/defdelegate_order.ex]: Updated in [9853a6e](https://github.com/nimblehq/compass-credo-plugin/pull/10/commits/9853a6ecac8307917297512ca42d711960509399)

File: priv/.credo.exs (hunk 1 above)
Comment (hanam1ni) [reply priv/.credo.exs]: Updated in [9853a6e](https://github.com/nimblehq/compass-credo-plugin/pull/10/commits/9853a6ecac8307917297512ca42d711960509399)

**Pull request merged**